tests/profiles_enabled/shipments/test_access_requests.py

```diff
@@ -0,0 +1,637 @@
+    def test_cant_request_own(self, client_alice, access_request_ro_attributes):
+        response = client_alice.post(self.list_url, access_request_ro_attributes)
+        AssertionHelper.HTTP_403(response, 'You do not have permission to perform this action.')
+
+    # approved/approved_at/approved_by,requester_id fields are read-only
+    def test_ro_fields(self, client_bob, user_alice_id, user_bob_id, access_request_ro_attributes, current_datetime):
+        response = client_bob.post(self.list_url, {**access_request_ro_attributes,
+                                                     **{'approved': True}})
+        AssertionHelper.HTTP_400(response, 'User does not have access to approve this access request')
+        response = client_bob.post(self.list_url, {**access_request_ro_attributes,
+                                                   **{'approved_at': current_datetime}})
+        AssertionHelper.HTTP_201(response, attributes={'approved_at': None})
+        response = client_bob.post(self.list_url, {**access_request_ro_attributes,
+                                                   **{'approved_by': user_alice_id}})
+        AssertionHelper.HTTP_201(response, attributes={'approved_by': None})
+        response = client_bob.post(self.list_url, {**access_request_ro_attributes,
+                                                   **{'requester_id': user_alice_id}})
+        AssertionHelper.HTTP_201(response, attributes={'requester_id': user_bob_id})
+
+    # Any authenticated user should be able to request access for any valid shipment ID (except their own)
+    def test_create_access_request(self, shipment_alice, client_bob, access_request_ro_attributes, user_bob_id):
+        response = client_bob.post(self.list_url, access_request_ro_attributes)
+        AssertionHelper.HTTP_201(response, entity_refs=AssertionHelper.EntityRef(
+            resource='AccessRequest',
+            attributes={**{'requester_id': user_bob_id}, **access_request_ro_attributes}
+        ))
+
+    # Multiple shipment access request creation
+    def test_create_multiple_access_requests(self, shipment_alice, client_bob, access_request_ro_attributes, access_request_rw_attributes, user_bob_id):
+        response = client_bob.post(self.list_url, access_request_ro_attributes)
+        AssertionHelper.HTTP_201(response, entity_refs=AssertionHelper.EntityRef(
+            resource='AccessRequest',
+            attributes={**{'requester_id': user_bob_id}, **access_request_ro_attributes}
+        ))
+
+        response = client_bob.post(self.list_url, access_request_rw_attributes)
+        AssertionHelper.HTTP_201(response, entity_refs=AssertionHelper.EntityRef(
+            resource='AccessRequest',
+            attributes={**{'requester_id': user_bob_id}, **access_request_rw_attributes}
+        ))
+
+    # TODO: Shipment owner(s?) should be notified of access requests (email, ???)
+
+    # Tracking and telemetry endpoints should not be able to be requested as RW
```

Comment:
Does the carrier wallet association on shipments allow carriers to update tracking/see shipment on their dashboard? Ex) Shipper A adds Carrier B (outside org) wallet on shipment creation. Would Carrier B be able to update tracking, or would they need to request access to the shipment in order to do so?

Reply:
The only person who is able to update tracking is whomever is able to sign and use the device that is associated with the shipment. The carrier/moderator wallet owners have read, update and delete permissions for their respective shipments shipments. In your example, carrier B would not require any additional calls after shipment creation to access or update the shipment.